pkg/common/cns-lib/vsphere/utils.go

```diff
@@ -460,3 +461,39 @@ func IsvSphereVersion70U3orAbove(ctx context.Context, aboutInfo types.AboutInfo)
 	// For all other versions.
 	return false, nil
 }
+
+// IsVolumeCreationSuspended checks whether a given Datastore has cns.vmware.com/datastoreSuspended customValue
+func IsVolumeCreationSuspended(datastoreInfo *DatastoreInfo, cnsMgrSuspendCreateVolumeFSS bool) bool {
```

Comment:
This is subjective, but this function shouldn't take in the flag `cnsMgrSuspendCreateVolumeFSS` IMO.
The check should be done at where this is being called.